salt/utils/azurearm.py

```diff
@@ -124,7 +124,7 @@ def _determine_auth(**kwargs):
 
     subscription_id = kwargs['subscription_id']
 
-    return credentials, subscription_id, cloud_env
+    return credentials, subscription_id.encode(), cloud_env
```

Comment:
This doesn't seem correct, since the documentation for the subscription ID in the azure mgmt modules says that it's supposed to be a `str`. Encoding would give you a bytestring, meaning this would likely break on Python 3. If what we need is a `str` type, then what we should do is:

1. Revert line 127 back to its original value.
2. Change line 125 to `subscription_id = kwargs['subscription_id']`

You'd also need to import `salt.utils.stringutils`.

Reply:
Hey, sorry I just realized I mistyped in my earlier comment. It should have been "Change line 125 to `subscription_id = salt.utils.stringutils.to_str(kwargs['subscription_id'])`."